setup.py

```diff
@@ -13,7 +13,7 @@ def get_version() -> str:
 
 install_requires = [
     "filelock",
-    "requests",
+    "requests>=2.27",
```

Comment:
What makes us require a very recent version of `requests`? This potentially creates dependency conflicts for users trying to install our library alongside other packages which potentially don't support latest `requests` versions. What's the oldest version we can work with here?

Reply:
Minimal required version of `requests` is 2.27.0 because [we use `JSONDecodeError`](https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/hf_api.py#L24), and it was introduced only in `2.27.0`: https://docs.python-requests.org/en/latest/community/updates/#id2

Reply:
When was `2.27.0` released? and do we know of libraries that pin an upper bound on the version of `requests` they use?

Reply:
`2.27.0`  was released on 2022-03-01

Reply:
`requests.JSONDecodeError` was introduced by https://github.com/psf/requests/pull/5856

Reply:
Ok, in my opinion this is ok to ship this version pin, given that most libraries would probably not pin an upper bound on the version of `requests` they use. And we can always relax this constraint if we realize it's affecting users?

Reply:
Let's merge this with this pinned version, I'll open a PR today to fix it and remove the constraint before we release a new version. The pinned version is way too recent. cc @LysandreJik